app/assets/javascripts/discourse/models/topic-list.js.es6

```diff
@@ -189,6 +189,20 @@ TopicList.reopenClass({
   // hide the category when it has no children
   hideUniformCategory(list, category) {
     list.set("hideCategory", !displayCategoryInList(list.site, category));
+  },
+
+  findOrResetCachedBy(filter) {
```

Comment:
One thing that we've started to move to over time is not using class methods on Ember objects anymore. They were useful back in the day when JS had no modules, but now that any file can export a named function it's easier to make it a pure function and export it that way.

Also, it would be nice if you could pass `session` as a param, then use `this.session` in your instance. It should be available that way!

Reply:
@eviltrout I put the function is a new file for cached topic list. I am really unsure where it should sit.